src/fiber/network.rs

```diff
@@ -3196,6 +3252,9 @@ where
         if let Err(err) = state.control.close().await {
             error!("Failed to close tentacle service: {}", err);
         }
+        debug!("Saving network actor state for {:?}", state.peer_id);
+        self.store
+            .insert_network_actor_state(&state.peer_id, state.persistent_state.clone());
```

Comment:
If I am going to add a store to the persistent_state, then there will be a cyclic dependency chain. The definition of the trait `NetworkActorStateStore` requires a `PersistentNetworkActorState` because we have to define what we want to save to/load from the store (this is the struct `PersistentNetworkActorState`). And the definition of  `PersistentNetworkActorState` contains the field `NetworkActorStateStore`, hence a dependency cycle. We should do something to ensure that we will not forget to save the state to persistent storage. This is the reason why I wrapped the function `save_peer_addresses` of `PersistentNetworkActorState` (which we should not call directly, we can only enforce this by rust's field/methods visibility) to a function `save_peer_addresses` of `NetworkActorState` (which does the final persistence work, and we should use this when possible).